app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/DefaultTasksRepository.kt

```diff
@@ -95,6 +98,13 @@ class DefaultTasksRepository(
 
     override suspend fun saveTask(task: Task) {
         coroutineScope {
+            // TODO Add comment here or change flow to talk about how to handle when remote fails
+            // In the case remote save fails, the fresh "local" data will be overwritten by the
+            // "old" remote data on next sync
+            // Same with all of the write operations below
+            // Simple flow could be updating the remote with a "loading" icon, waiting to see it if
+            // succeeded before saving in local - on failure, inform user and make no change to local
```

Comment:
We simply don't have that feature. It would complicate things a lot. We would have to show a loading indicator while the user uses the local data. What if it changes before the data comes? This kind of synchronization is hard to do and I'd use something like Firestore to deal with it. It's explained here https://www.youtube.com/watch?v=xHXn3Kg2IQE

Happy to add a comment, but it would apply to the master branch!

Reply:
+1 let's add a comment about why this is hard --- is there a talk that is more current (this is talking about content providers and sync adapters) about handling this problem?